backend/amd64/emit.mlp

```diff
@@ -944,7 +944,11 @@ let emit_instr fallthrough i =
        semaphores are of type unsigned short.
        [1] https://sourceware.org/systemtap/wiki/UserSpaceProbeImplementation
     *)
-    I.mov (addressing (Ibased(semaphore_sym, 0)) WORD i 0) (res16 i 0);
+    (* OCaml has it's own semaphore, in addition to system tap,
+       to control ocaml probe handlers independently from stap probe handlers.
+       It is placed immediately after stap semaphore, and is the same
+       size - hence offset 2. *)
```

Comment:
I don't understand why this is offset 2 not 1.  Is this value not in units of 8-byte words?

Reply:
The units are bytes, and the size of the new semaphore is set to be the same as the size of stap semaphores, so the code for reading them remains unchanged.

Reply:
ah, and the semaphores are 16-bit quantities.  ok.